### PR TITLE
Make tarpc request execution sequential for a single connection

### PR DESCRIPTION
**Pull request by paullegranddc**
# What does this PR do?

A brief description of the change being made with this pull request.

# Motivation

What inspired you to submit this pull request?

# Additional Notes

Anything else we should know when reviewing?

# How to test the change?

Describe here in detail how the change can be validated.

### CODE REVIEW

**Review by pawelchcki — APPROVED** [review]
OOhh - magic!

:shipit:

File: ipc/src/sequential.rs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+            match response_handler {
+                Ok(resp) => {
+                    let server = self.serve.clone();
+                    if let Err(err) = self.as_ref().tx.try_send((server, resp)) {
+                        // TODO: should we log something in case we drop the request on the floor?
+                    }
+                }
+                Err(e) => {
+                    // TODO: should we log something in case we drop the request on the floor?
+                    break;
```
[the remark this answers is not in the document]
Comment (pawelchcki) [reply ipc/src/sequential.rs:82]: TODO is ok for now. But we need some introspection into sidecar in general - e.g. being able to report back telemetry about itself. Dropped requests - etc

**Comment by bwoebi** [comment]
@paullegranddc I just tried https://github.com/DataDog/libdatadog/commit/4b955be4af27b92edc7b2973751476b48aca850a - this seems to fix the hangs, can you please verify / include in this PR? Thanks!

**Review by pawelchcki — APPROVED** [review]

**Comment by bwoebi** [comment]
Superseded by #161

**Pull request closed without merging**